v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSourceDbIT.java

```diff
@@ -0,0 +1,235 @@
+    spannerMetadataResourceManager = createSpannerMetadataDatabase();
+
+    // Run a simple IT
+    simpleGcsToSourceDbTest(
+        testName,
+        TABLE,
+        mySQLSource,
+        DatastreamResourceManager.DestinationOutputFormat.AVRO_FILE_FORMAT,
+        b -> b.addParameter("inputFileFormat", "avro"));
+  }
+
+  private SpannerResourceManager createSpannerDatabase() throws IOException {
+    SpannerResourceManager spannerResourceManager =
+        SpannerResourceManager.builder("rr-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String ddl =
+        String.join(
+            " ",
+            Resources.readLines(
+                Resources.getResource(SPANNER_DDL_RESOURCE), StandardCharsets.UTF_8));
+    ddl = ddl.trim();
+    String[] ddls = ddl.split(";");
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    return spannerResourceManager;
+  }
+
+  private SpannerResourceManager createSpannerMetadataDatabase() throws IOException {
+    SpannerResourceManager spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    return spannerMetadataResourceManager;
+  }
+
+  private void simpleGcsToSourceDbTest(
+      String testName,
+      String tableName,
+      JDBCSource jdbcSource,
+      DatastreamResourceManager.DestinationOutputFormat fileFormat,
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder)
+      throws IOException, InterruptedException {
+
+    createAndUploadShardConfigToGcs();
+    gcsClient.uploadArtifact(
+        "input/session.json", Resources.getResource(SESSION_FILE_RESOURSE).getPath());
+    launchWriterDataflowJob();
+
+    // Write events to GCS
+    gcsClient.uploadArtifact(
+        "cdc/events.txt", Resources.getResource("GCSToSourceDbIT/events.txt").getPath());
+
+    // Assert events on Mysql
+    long rowCount = 0;
+    for (int i = 0; rowCount != 1 && i < 60; ++i) {
+      rowCount = jdbcResourceManager.getRowCount(TABLE);
+      LOG.info("Row count = {}, Waiting for 30s if row count not = 1", rowCount);
+      Thread.sleep(10000);
+    }
+    assertThat(rowCount).isEqualTo(1);
+    List<Map<String, Object>> rows = jdbcResourceManager.readTable(TABLE);
+    assertThat(rows).hasSize(1);
+    assertThat(rows.get(0).getOrDefault("id", -1)).isEqualTo(1);
+    assertThat(rows.get(0).getOrDefault("name", "non existent")).isEqualTo("FF");
+  }
+
+  private void launchWriterDataflowJob() throws IOException {
+    // default parameters
+    Map<String, String> params =
```

Comment:
If the reader template is not there, the writer template needs to know the start time and the window duration. It needs that to check for the GCS files. The GCSReader class has the logic to construct the file name.